src/Provider/AbstractProvider.php

```diff
@@ -98,7 +108,15 @@ public function __construct($options = [], HttpAdapterInterface $httpClient = nu
             }
         }
 
-        $this->setHttpClient($httpClient ?: new CurlHttpAdapter());
+        if (empty($collaborators['httpClient'])) {
```

Comment:
The difference being that `false` would be detected as empty and the default adapter would be injected. I think it's really bad practice to use `!isset` instead of `empty`, unless you want to explicitly allow for `false` or `0` as a value. Way too often I see this kind of thing:

```php
if (isset($something) && $something !== false) { ... }
// can be replaced with a much more clear intent
if (!empty($something)) { ... }
```